src/notification.js

```diff
@@ -0,0 +1,322 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var common = require('@google-cloud/common');
+var is = require('is');
+var util = require('util');
+
+/**
+ * A Notification object is created from your {@link Bucket} object using
+ * {@link Bucket#notification}. Use it to interact with Cloud Pub/Sub
+ * notifications.
+ *
+ * @see [Cloud Pub/Sub Notifications for Google Cloud Storage]{@link https://cloud.google.com/storage/docs/pubsub-notifications}
+ *
+ * @class
+ * @hideconstructor
+ *
+ * @param {Bucket} bucket The bucket instance this notification is attached to.
+ * @param {string} id The ID of the notification.
+ *
+ * @example
+ * var storage = require('@google-cloud/storage')();
+ * var myBucket = storage.bucket('my-bucket');
+ *
+ * var notification = myBucket.notification('1');
+ */
+function Notification(bucket, id) {
+  var methods = {
+    /**
+     * Creates a notification subscription for the bucket.
+     *
+     * @see [Notifications: insert]{@link https://cloud.google.com/storage/docs/json_api/v1/notifications/insert}
+     *
+     * @param {Topic|string} topic The Cloud PubSub topic to which this
+     *     subscription publishes. If the project ID is omitted, the current
+     *     project ID will be used.
+     *
+     *     Acceptable formats are:
+     *     - `projects/grape-spaceship-123/topics/my-topic`
+     *
+     *     - `my-topic`
+     * @param {Bucket~CreateNotificationRequest} [options] Metadata to set for
```

Comment:
Was this required for JSDoc to link to the right symbol?

Reply:
Okay, I just tested it. Need to remove the `Bucket~`, and for the documentation of `Bucket#create`, need to remove the `Storage~`.